### ISSUE TITLE
Webpack build/serve compressed bundle files

### ISSUE THREAD

**Issue by PeterBreen** [DemocracyLab/CivicTechExchange#968]
We use webpack to create three js bundle files on build:
 - runtime.bundle.js
 - main.bundle.js
 - vendors.bundle.js 

See PR #346 for context.

These are served via straight hyperlinks on the one html file which django serves and react uses as a base for everything else:
https://github.com/DemocracyLab/CivicTechExchange/blob/7d28862876c0142e0dc0f8957fd44b76b3a1b7eb/civictechprojects/templates/new_index.html#L16-L18

However, we are not using gzip or brotli or any compression. According to heroku, [that's our app's responsibility](https://devcenter.heroku.com/articles/compressing-http-messages-with-gzip) and we're getting rightly knocked in Lighthouse reporting for not doing this. 
![image](https://user-images.githubusercontent.com/16870466/236045436-b2ea3b60-86db-49ca-bf03-9df18730af28.png)

Therefore: in order to reduce initial load time and transfer, we should offer compressed bundles. Gzip, at minimum. Webpack should be able to build them as part of `npm run build` and then we need to offer the compressed version to any browser requesting it with the uncompressed bundles as fallback options. Right now the app doesn't care if an http request has `Accept-Encoding:` because we don't have any encoded bundles to offer and don't have a way to send them since the hrefs are just hrefs.

As you can see in the lighthouse report I posted our compiled sass styles.css file is also something we can offer this way and I'd like to see that too, but the bundles are a higher priority.

To sum up,
 - configure webpack to produce compressed bundles on build, as well as uncompressed
 - serve these if the http request has the appropriate accept-encoding header
 - serve uncompressed bundles as fallback if needed
 - don't break anything ;)

**Comment by QuanGiap** [comment]
From my research, we can use [compression-webpack-plugin](https://www.npmjs.com/package/compression-webpack-plugin) to compress the bundle files with Gzip.

**Comment by ddfridley** [comment]
@abnud1 Hi, I'm helping with development focus for DemocarcyLab. We've created a [DemocracyLab Dashboard](https://github.com/orgs/DemocracyLab/projects/1/views/1?layout=board) and would like to update the status of this issue.

There seems to be an outstanding question in the PR #1029 
> Just so I'm clear - would this require changing how we serve files on production (needing gunicorn instead of django's runserver) as well, or is that a dev environment specific change?

Are you still working on it? If so, what are the next steps, or what challenges are there to moving forward?
Or is this still an issue/should we close it?
Or, if you just aren't available to work on it anymore, please let me know, and/or unassign yourself
Or something else?

Thanks!
